src/openms/include/OpenMS/IONMOBILITY/MSRunIMSplitter.h

```diff
@@ -0,0 +1,67 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2020.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Eugen Netz $
+// $Authors: Eugen Netz $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/KERNEL/StandardTypes.h>
+#include <vector>
+
+namespace OpenMS
+{
+
+    /**
+      @brief This class splits PeakMaps by IM values
+
+      Contains a function to split a PeakMap by FAIMS compensation voltages.
+    */
+    class OPENMS_DLLAPI MSRunIMSplitter
+    {
+    public:
+      virtual ~MSRunIMSplitter() {}
+
+      /**
+        @brief Splits a PeakMap into one PeakMap per FAIMS CV value
+
+        This only works with a PeakMap that has a FAIMS compensation voltage
+        associated with each spectrum.
+        The spectra from the original PeakMap are moved to new PeakMaps,
+        so the orinǵinal PeakMap is unusable afterwards.
+
+        @param exp The PeakMap
+      */
```

Comment:
this function should IMHO throw an Exception if there is no FAIMS data available.

Maybe add a function: `bool isFAIMSExperiment(const PeakMap& exp)` which users can call beforehand to make sure its safe.